packages/jest-runtime/src/index.ts

```diff
@@ -1640,6 +1667,50 @@ export default class Runtime {
     return evaluateSyntheticModule(module);
   }
 
+  private async _importWasmModule(
+    source: Buffer,
+    identifier: string,
+    context: VMContext,
+  ) {
+    const wasmModule = await WebAssembly.compile(source);
+
+    const exports = WebAssembly.Module.exports(wasmModule);
+    const imports = WebAssembly.Module.imports(wasmModule);
+
+    const moduleLookup: Record<string, VMModule> = {};
+    for (const {module} of imports) {
+      if (moduleLookup[module] === undefined) {
+        moduleLookup[module] = await this.linkAndEvaluateModule(
```

Comment:
should these be put into `this. _esmoduleRegistry` as well?

Reply:
I’ve replaced `linkAndEvaluateModule` with `loadEsmModule` in https://github.com/facebook/jest/pull/13505/commits/26db3fde3eb67fa82cd0ff8e7de19a74bdfff189. Not 100% sure if this is right, but AFAIU, `loadEsmModule` is a wrapper function that deals with `_esmoduleRegistry`. If that’s not the right solution, feel free to push to my branch if the required change is not trivial.

Reply:
Hmmm I should have kept `linkAndEvaluateModule`. Using `loadEsmModule` here can result

```
TypeError: modulePath.endsWith is not a function
  at isWasm (../../../node_modules/jest-runtime/build/index.js:210:41)
```

This happens when Wasm imports refer back to a javascript file that calls it. E.g.:

```
imports: [
  {
    module: './index_bg.js',
    name: '__wbindgen_json_parse',
    kind: 'function'
  },
  {
    module: './index_bg.js',
    name: '__wbindgen_json_serialize',
    kind: 'function'
  },
  {
    module: './index_bg.js',
    name: '__wbindgen_throw',
    kind: 'function'
  }
]
```

`loadEsmModule` returns 

```
modulePath: SourceTextModule {
  status: 'linking',
  identifier: '/path/to/index_bg.js',
  context: {
    clearInterval: [Function: clearInterval],
    clearTimeout: [Function: clearTimeout],
    setInterval: [Function: setInterval],
    setTimeout: [Function: setTimeout] {
      [Symbol(nodejs.util.promisify.custom)]: [Getter]
    },
```

which is then mistakenly passed to to `isWasm` instead of a string.

I can try submiting a follow-up PR with a fix in a few days.

Reply:
Thanks!

Reply:
https://github.com/facebook/jest/pull/13608